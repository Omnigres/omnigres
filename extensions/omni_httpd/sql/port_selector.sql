--- Force omni_httpd to select a port
BEGIN;
WITH listener AS (INSERT INTO omni_httpd.listeners (address, port, handler_id) VALUES ('127.0.0.1', 0, (SELECT id FROM handler)) RETURNING id),
     handler AS (INSERT INTO omni_httpd.handlers VALUES (DEFAULT) RETURNING id)
     SELECT * FROM listener;
DELETE FROM omni_httpd.configuration_reloads;
END;

CALL omni_httpd.wait_for_configuration_reloads(1);

-- Ensure port was updated
SELECT count(*) FROM omni_httpd.listeners WHERE port = 0;

-- TODO: Test request on a given port. Needs non-shell http client