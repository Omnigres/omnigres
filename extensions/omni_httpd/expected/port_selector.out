--- Force omni_httpd to select a port
BEGIN;
WITH listener AS (INSERT INTO omni_httpd.listeners (address, port, handler_id) VALUES ('127.0.0.1', 0, (SELECT id FROM handler)) RETURNING id),
     handler AS (INSERT INTO omni_httpd.handlers VALUES (DEFAULT) RETURNING id)
     SELECT * FROM listener;
ERROR:  relation "handler" does not exist
LINE 1: ...ndler_id) VALUES ('127.0.0.1', 0, (SELECT id FROM handler)) ...
                                                             ^
DETAIL:  There is a WITH item named "handler", but it cannot be referenced from this part of the query.
HINT:  Use WITH RECURSIVE, or re-order the WITH items to remove forward references.
DELETE FROM omni_httpd.configuration_reloads;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
END;
CALL omni_httpd.wait_for_configuration_reloads(1);
-- Ensure port was updated
SELECT count(*) FROM omni_httpd.listeners WHERE port = 0;
 count 
-------
     0
(1 row)

-- TODO: Test request on a given port. Needs non-shell http client
